datafusion/core/tests/sql/window.rs

```diff
@@ -2290,30 +2290,32 @@ async fn test_remove_unnecessary_sort_in_sub_query() -> Result<()> {
     let dataframe = ctx.sql(sql).await.expect(&msg);
     let physical_plan = dataframe.create_physical_plan().await?;
     let formatted = displayable(physical_plan.as_ref()).indent().to_string();
+
     // Unnecessary Sort in the sub query is removed
-    let expected = {
-        vec![
-            "ProjectionExec: expr=[COUNT(UInt8(1))@0 as global_count]",
-            "  AggregateExec: mode=Final, gby=[], aggr=[COUNT(UInt8(1))]",
-            "    CoalescePartitionsExec",
-            "      AggregateExec: mode=Partial, gby=[], aggr=[COUNT(UInt8(1))]",
-            "        RepartitionExec: partitioning=RoundRobinBatch(8)",
```

Comment:
Using https://github.com/apache/arrow-datafusion/pull/4885/files?w=1 is easiest to see what changed -- the final group by with no group exprs is no longer done in parallel which I think will actually improve performance

Reply:
I am worried about this change. Looking into it

Reply:
fixed